yajsapi/rest/common.ts

```diff
@@ -0,0 +1,55 @@
+import { Callable, logger, sleep } from "../utils";
+
+export function is_intermittent_error(e) {
+  if (e.response && e.response.status === 408) { return true; }
+  if (e.code === "ECONNABORTED" && e.message && e.message.includes("timeout")) { return true; }
+  if (e.code === "ETIMEDOUT") { return true; }
+  return false;
+}
+
+export async function suppress_exceptions(
+  condition: Callable<Error, boolean>,
+  block: Callable<void, any>,
+  report_exceptions: boolean = true
+): Promise<any> {
+  try {
+    return await block();
+  } catch (error) {
+    if (condition(error)) {
+      logger.debug(`Exception suppressed: ${error}`);
+    } else {
+      throw error;
+    }
+  }
+}
+
+export async function repeat_on_error(
+  block: Callable<void, any>,
+  max_tries: number = 5,
+  max_duration_ms = 15000,
+  interval: number = 0.0,
+  condition: Callable<Error, boolean> = is_intermittent_error
+) {
+  let start_time = Date.now();
+  for (let try_num = 1; try_num <= max_tries; ++try_num) {
+    if (try_num > 1) {
+      await sleep(interval);
```

Comment:
Shouldn't we sleep for `min(interval, start_time + max_duration_ms - Date.now())` here?

Reply:
Yes, then the `max_duration_ms` name would be more relevant.

Reply:
Fixed